addons/pos_self_order/models/res_config_settings.py

```diff
@@ -52,12 +52,19 @@ def _onchange_pos_self_order_kiosk(self):
             self.is_kiosk_mode = True
             self.pos_module_pos_restaurant = False
             self.pos_self_ordering_pay_after = "each"
+            cash_payment_methods = self.pos_payment_method_ids.filtered(lambda x: x.is_cash_count)
+            self.pos_payment_method_ids = self.pos_payment_method_ids - cash_payment_methods
         else:
             self.is_kiosk_mode = False
 
             if not self.pos_module_pos_restaurant:
                 self.pos_self_ordering_service_mode = 'counter'
 
+    @api.onchange("pos_payment_method_ids")
+    def _onchange_pos_payment_method_ids(self):
+        if self.pos_self_ordering_mode == 'kiosk' and any(pm.is_cash_count for pm in self.pos_payment_method_ids):
+            raise ValidationError(_("You cannot add cash payment methods in kiosk mode."))
+
```

Comment:
what if you override `payment_method_ids` and add a `domain` property such that the `cash` options are not even displayed? Then you won't need this `onchange` at all. Why display the `cash` options in the dropdown if you are not actually allowed to select them?

Reply:
That's another good solution to me but I think it might be strange for users not to see all their payment methods compared to the onchange error. They might not understand why they're not there.

Again up to you @SalukiMakingCode 

Maybe ask @dama-odoo ?

Reply:
I agree that's a ingenious trick. But I also think it might be strange for the users not to see all their payment method.

@dama-odoo what do you think of that ?

Reply:
I discussed with @dama-odoo and he agreed with the domain approach. Let's try to do that